packages/vscode-extension/src/handlers/updateActionWithMCP.ts

```diff
@@ -46,15 +85,17 @@ export async function updateActionWithMCP(args?: any[]): Promise<Result<any, FxE
   let mcpName = args && args.length > 0 ? args[0].serverName : undefined;
   let server = args && args.length > 0 ? args[0].serverConfig?.url : undefined;
   let isLocalMCP = args && args.length > 0 && args[0].serverConfig?.type === "stdio";
-  // For stdio type (local MCP), extract identifier from args array (e.g., args[2] contains the identifier)
-  let localServerIdentifier =
-    args && args.length > 0 ? extractLocalServerIdentifier(args[0].serverConfig) : undefined;
 
   // Sanitize mcpName if it's provided as an argument
   if (mcpName) {
     mcpName = sanitizeMCPName(mcpName);
   }
 
+  let localServerIdentifier =
+    args && args.length > 0 && isLocalMCP
```

Comment:
isLocalMCP already checks args
```suggestion
    isLocalMCP
```